wandb/beta/workflows.py

```diff
@@ -0,0 +1,266 @@
+) -> "_SavedModel":
+    """Logs a model object to enable model-centric workflows in the UI.
+    Supported frameworks include PyTorch, Keras, Tensorflow, Scikit-learn, etc.
+    Under the hood, we create a model artifact, bind it to the run that produced this model,
+    associate it with the latest metrics logged with `wandb.log(...)` and more.
+
+    Args:
+        model_obj: any model object created with the following ML frameworks: PyTorch, Keras, Tensorflow, Scikit-learn.
+        name: `str` - name of the model artifact that will be created to house this model_obj.
+        aliases: `str, List[str]` - optional alias(es) that will be applied on this model and allow for unique
+            identification. The alias "latest" will always be applied to the latest version of a model.
+        description: `str` - text description/notes about the model - will be visible in the Model Card UI.
+        metadata: `Dict` - model-specific metadata goes here - will be visible the UI.
+        project: `str` - project under which to place this artifact.
+        scope_project: `bool` - If true, name of this model artifact will not be suffixed by `-{run_id}`.
+
+    Returns:
+        _SavedModel instance
+
+    Example:
+        import torch.nn as nn
+        import torch.nn.functional as F
+
+        class Net(nn.Module):
+            def __init__(self):
+                super(Net, self).__init__()
+                self.fc1 = nn.Linear(10, 10)
+
+            def forward(self, x):
+                x = self.fc1(x)
+                x = F.relu(x)
+                return x
+
+        model = Net()
+        sm = log_model(model, "my-simple-model", aliases=["best"])
+
+    """
+    model = data_types._SavedModel.init(model_obj, **kwargs)
+    _ = _log_artifact_version(
+        name=name,
+        type="model",
+        entries={"index": model,},
+        aliases=aliases,
+        description=description,
+        metadata=metadata,
+        project=project,
+        scope_project=scope_project,
+        job_type="log_model",
+    )
+    # TODO: handle offline mode appropriately.
+    return model
+
+
+def use_model(aliased_path: str) -> "_SavedModel":
+    """Allows the user to fetch their saved model with an alias.
+    Under the hood, we use the alias to fetch the model artifact containing the serialized model files
+    and rebuild the model object from these files. We also declare the fetched model artifact as an
+    input to the run (with `run.use_artifact`).
+
+    Args:
+        aliased_path: `str` - the following forms are valid: "name:version", "name:alias". May be prefixed with "entity/project".
+
+    Returns:
+        _SavedModel instance
+
+    Example:
+        # assuming you have previously logged a model with the name "my-simple-model"
+        sm = use_model("my-simple-model:latest")
+        model = sm.model_obj()
+    """
+    parts = aliased_path.split(":")
+    if len(parts) == 1:
+        aliased_path += "latest"
```

Comment:
wait do you need `:` between the path and the `latest`? 
```
use_model("my-simple-model")
```
will result in: `aliased_path = my-simple-modellatest`

from the description i'm not sure if this is a valid name...

Reply:
probably good to add a simple test here

Reply:
Ahh, great catch! Let me add this in the next commit. I think we want to make `:` required for `use_model` for now; let me raise an error otherwise. This may change in the future with user behavior; we may just allow the user to specify a name without an alias and we'll just pull either `best` if it exists or `latest`. Something like that.